.github/workflows/build.yml

```diff
@@ -1,39 +1,73 @@
-name: Build
+name: Build with Kubernetes
 
 env:
   MAVEN_ARGS: -V -ntp -e
 
 on:
   workflow_call:
+    inputs:
+      kube-version:
+        type: string
+        required: true
 
 jobs:
+  set_up_kubernetes:
+    name: Set up Kubernetes ${{ inputs.kube-version }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Minikube
+        uses: manusa/actions-setup-minikube@v2.13.1
+        with:
+          minikube version: 'v1.34.0'
+          kubernetes version: '${{ inputs.kube-version }}'
+          driver: 'docker'
+          github token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Save minikube directory
+        id: minikube
+        run: |
+          echo "minikube-dir=$MINIKUBE_HOME" >> $GITHUB_OUTPUT
+
+      - name: Upload minikube
+        uses: actions/upload-artifact@v4
+        with:
+          name: minikube-${{ inputs.kube-version }}
+          path: ${{ steps.minikube.outputs.minikube-dir }}
+          include-hidden-files: true
+
   integration_tests:
+    name: "Integration tests (${{ inputs.kube-version }}, ${{ matrix.java }}, ${{ matrix.it-category }})"
```

Comment:
Same here, as we never call this workflow directly, the kube version is redundant and leaving it out would make the rest stand out more, imo.

Reply:
So I'm not sure if it's better like this, but at least the information is not repeated.